api/wasm/cpp/proxy_wasm_intrinsics.h

```diff
@@ -10,7 +10,7 @@
 #endif
 
 #include <cstdint>
-using EnumType = uint32_t;
+using EnumType = int32_t;
```

Comment:
Why are you making this change? For the Invalid as "-1"? You could have replace it by "std::numeric_limits<EnumType>::max()"

Reply:
I am matching the envoy GrpcStatus class so we can use that one when using the null VM.